storage/mysql/map_storage.go

```diff
@@ -124,21 +124,22 @@ func (m *mapTX) Set(keyHash []byte, value trillian.MapLeaf) error {
 	}
 	defer stmt.Close()
 
-	_, err = stmt.Exec(m.ms.mapID, []byte(keyHash), m.writeRevision, flatValue)
+	_, err = stmt.Exec(m.ms.mapID, keyHash, m.writeRevision, flatValue)
 	return err
 }
 
-func (m *mapTX) Get(revision int64, keyHashes [][]byte) ([]trillian.MapLeaf, error) {
-	stmt, err := m.ms.getStmt(selectMapLeafSQL, len(keyHashes), "?", "?")
+// TODO: API problem: there's nothing to say which map leaves correspond to which indexes.
+func (m *mapTX) Get(revision int64, indexes [][]byte) ([]trillian.MapLeaf, error) {
+	stmt, err := m.ms.getStmt(selectMapLeafSQL, len(indexes), "?", "?")
 	if err != nil {
 		return nil, err
 	}
 	stx := m.tx.Stmt(stmt)
 	defer stx.Close()
 
-	args := make([]interface{}, 0, len(keyHashes)+2)
-	for _, k := range keyHashes {
-		args = append(args, []byte(k[:]))
+	args := make([]interface{}, len(indexes), len(indexes)+2)
+	for i, index := range indexes {
```

Comment:
It _might_ be marginally faster, although the benchmark I made (https://play.golang.org/p/FYh5arq6D9) is not clear cut on the matter: 4 runs:
```
BenchmarkDirect-12    	1000000000	         3.28 ns/op
BenchmarkAppend-12    	2000000000	         4.42 ns/op
BenchmarkDirect-12    	1000000000	         4.27 ns/op
BenchmarkAppend-12    	2000000000	         3.15 ns/op
BenchmarkDirect-12    	1000000000	         3.38 ns/op
BenchmarkAppend-12    	1000000000	         1.98 ns/op
BenchmarkDirect-12    	1000000000	         3.00 ns/op
BenchmarkAppend-12    	1000000000	         2.00 ns/op
```

But even if it were faster (and I suspect it could only ever be marginally faster since both approaches have to do bounds checking) you're adding complexity to pay for it; in this case an extra variable, and in other places where you've done the same thing you've added an extra explicit `var` line to create the variable, as well as an extra explicit `i++` because you're iterating over a map.   But, then, the next thing which happens is many orders of magnitude slower (because it goes to a remote DB server, and then possibly to disk, so your complexity down-payment is for nothing.

`append` is a fundamental part of the language - it's a built-in keyword, it doesn't get more idiomatic than that :)